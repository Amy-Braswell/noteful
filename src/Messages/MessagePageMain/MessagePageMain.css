
.MessagePageMain {
    background-color: #efefef;
    height: 100%;
    padding: 10px;
    width: 100%;
  }
  
.MessagePageMain__content {
  color: #3a3939;
  max-width: 380px;
}

#root > div > main > section > button {
  background-color: #dddddd;
  padding: 5px 15px;
}

#root > div > main > section > button > a{
  text-decoration: none;
  color:#0b1f3a;
} 