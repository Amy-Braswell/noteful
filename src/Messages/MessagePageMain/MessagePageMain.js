import React from 'react'
import ApiContext from '../../ApiContext'
import { findMessage } from '../../messages-helpers'
import CircleButton from '../../CircleButton/CircleButton'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Message from '../Message/Message'
import './MessagePageMain.css'

export default class MessagePageMain extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { 
        window.history.back()
       }
    },
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  handleDeleteMessage = messageId => {
    this.props.history.push(`/`)
  }

  render(){
    const { messages=[] } = this.context
    const { messageId } = this.props.match.params
    const message = findMessage(messages, messageId) || { content: '' }
    return(
      <section className='MessagePageMain'>
        <CircleButton
          tag='button'
          role='link'
          onClick={() => this.props.history.goBack()}
          className='PostPageNav__back-button'
        >
          <FontAwesomeIcon icon={['fa', 'chevron-left']} />
          <br />
          Back
        </CircleButton>
        <Message
          id={message.id}
          name={message.name}
          modified={message.modified}
          onDeleteMessage={this.handleDeleteMessage}
        />

        <div className='MessagePageMain__content'>
          {message.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>{para}</p>
          )}
        </div>
      </section>
    )
  }
}