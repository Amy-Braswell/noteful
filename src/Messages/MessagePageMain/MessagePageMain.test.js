import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MessagePageMain from './MessagePageMain'

describe(`MessagePageMain component`, () => {
  it('renders a .MessagePageMain by default', () => {
    const wrapper = shallow(<MessagePageMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip('renders a Message with message prop', () => {
    const props = {
      match: {
        params: {
          messageId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }
    const context = {
      messages: [{
        id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
        name: `Dogs`,
        modified: `2019-01-03T00:00:00.000Z`,
        // memberid: b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1,
        content: "Corporis accusamus placeat.\n \rUnde."
      }]
    }
    const message = shallow(<MessagePageMain {...props} />, context)
      .find('Message')
    expect(toJson(message)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    const props = {
      match: {
        params: {
          messageId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }

    const messagesContextWithDifferentContent = [
      {
        messages: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            content: "Content with n r.\n \rafter n r.",
          }
        ]
      },
      {
        messages: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            content: "Content with n.\nafter."
          }
        ]
      }
    ]

    messagesContextWithDifferentContent.forEach(context => {
      const content = shallow(<MessagePageMain {...props} />, context)
        .find('MessagePageMain__content')
      expect(toJson(content)).toMatchSnapshot()
    })
  })
})