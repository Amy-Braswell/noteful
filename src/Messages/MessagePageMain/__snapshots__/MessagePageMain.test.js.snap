// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagePageMain component renders a .MessagePageMain by default 1`] = `
<section
  className="MessagePageMain"
>
  <NavCircleButton
    className="PostPageNav__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fa",
          "chevron-left",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavCircleButton>
  <Message
    onDeleteMessage={[Function]}
  />
  <div
    className="MessagePageMain__content"
  >
    <p
      key="0"
    />
  </div>
</section>
`;
